src/sqlfluff/dialects/dialect_mysql.py

```diff
@@ -776,6 +793,22 @@ class IntervalExpressionSegment(BaseSegment):
 )
 
 
+mysql_dialect.insert_lexer_matchers(
+    [
+        StringLexer("ampersand_ampersand", "&&", CodeSegment),
```

Comment:
We call them `double_quotes` so think we should do same here.

```suggestion
        StringLexer("double_ampersand", "&&", CodeSegment),
```